src/modules/claims/claims.service.ts

```diff
@@ -1494,12 +1505,32 @@ export class ClaimsService {
     if (!proofVerified) {
       errors.push(ERROR_MESSAGES.PROOF_NOT_VERIFIED);
     }
+    let isExpired = false;
+    if (payload?.exp) {
+      if (this.claimIsExpired(payload.exp)) {
+        isExpired = true;
+        errors.push(ERROR_MESSAGES.CLAIM_EXPIRED);
+      }
+    }
     return {
       errors: errors,
-      isVerified: !!proofVerified && !!issuerVerified,
+      isVerified: !!proofVerified && !!issuerVerified && !isExpired,
     };
   }
 
+  async fetchCredential(
+    subjectDID: string,
+    roleNamespace: string
+  ): Promise<
+    VerifiableCredential<RoleCredentialSubject> | RoleEIP191JWT | undefined
+  > {
+    const resolvedCredential = await this._credentialResolver.getCredential(
+      subjectDID,
+      roleNamespace
+    );
+    return resolvedCredential;
```

Comment:
It's a bit of personal preference, but it doesn't do much to have the variable in this case. If you like having it, feel free to keep it though.
```suggestion
    return await this._credentialResolver.getCredential(
      subjectDID,
      roleNamespace
    );
```